dali/image/jpeg.h

```diff
@@ -15,28 +15,37 @@
 #ifndef DALI_IMAGE_JPEG_H_
 #define DALI_IMAGE_JPEG_H_
 
+#include <utility>
+#include <memory>
+
+#ifdef DALI_USE_JPEG_TURBO
+
+#include <turbojpeg.h>
+
+#endif  // DALI_USE_JPEG_TURBO
+
 #include "dali/common.h"
-#include "dali/error_handling.h"
-#include "dali/pipeline/data/backend.h"
-#include "dali/pipeline/data/tensor.h"
+#include "dali/image/generic_image.h"
 
 namespace dali {
 
-/**
- * @brief Returns 'true' if input compressed image is a jpeg
- */
-DLL_PUBLIC bool CheckIsJPEG(const uint8 *jpeg, int size);
-
-/**
- * @brief Gets the dimensions of the jpeg encoded image
- */
-DLL_PUBLIC DALIError_t GetJPEGImageDims(const uint8 *jpeg, int size, int *h, int *w);
-
-/**
- * @brief Decodes `jpeg` into the the buffer pointed to by `image`
- */
-DLL_PUBLIC DALIError_t DecodeJPEGHost(const uint8 *jpeg, int size,
-    DALIImageType image_type, Tensor<CPUBackend>* output);
+class JpegImage : public GenericImage {
```

Comment:
I could be a good idea to mark the overrides from GenericImage as final

Reply:
Done